### ISSUE TITLE
Annotate Synapse submission entities with Tool information

### ISSUE THREAD

**Issue by tschaffter** [nlpsandbox/nlpsandbox-controller#77]
### Is your proposal related to a problem?

I would like to choice the license of tools in the leaderboards. This will give an indication to the user before he/she attempts to pull the image. We could also use this information to pick a tool that we can feature with the PHI Deidentifier. We could also in the future create/filter leaderboards to only show tools with open licenses.

### Describe the solution you'd like

During the evaluation workflow of a tool:

1. Call `/tool` to get information about the tool (potentially `/tool/dependencies` too)
2. Annotate the Synapse submission entity (consider using a prefix for all the properties of the Tool object)

**Comment by thomasyu888** [comment]
@tschaffter, getting `/tool/dependencies` needs to be added to the client prior to being used.

The prefix would work except there are two fields in the `Tool` object that already have `tool` as the prefix (`toolType`, and `toolApiVersion`.  What did you have in mind for a prefix?

**Comment by tschaffter** [comment]
> The prefix would work except there are two fields in the Tool object that already have tool as the prefix (toolType, and toolApiVersion. What did you have in mind for a prefix?

I remember that this scenario crossed my mind. What about the prefix `tool__` (two underscores)?

> You can see an example of this annotation being added if you scroll really to the right. 

This looks great!

> What did want to show in the leaderboard?

- Tool name
- Tool version
- License
- Description
- Url

**Comment by thomasyu888** [comment]
Thanks @tschaffter , so the two conflicting fields will be `tool__tool_api_version`, `tool__tool_name`?

**Comment by tschaffter** [comment]
1. I agree with renaming `toolType` to `type` and `toolApiVersion` to `apiVersion`. In general I avoided to use the keyword `type` in schemas for no strong reason other that "type" may be a reserved keyword in some programming languages, which I tried to avoid. I can find APIs that meet with both approach. Exploring the [API of GitHub](https://raw.githubusercontent.com/github/rest-api-description/main/descriptions/api.github.com/api.github.com.json ) lead me to accept the use of the "type".

2. The proposed prefix "tool__" would be a way to structure the annotations of a Synapse entity and conserve the hierarchy of information. For example Tool: { plop: { a = 1 }} could be encoded as the annotation `tool__plop__a = 1`. Since these variable are not exposed directly to the frontend user, I think that it would be fine using double underscores.
